### DISCUSSION TITLE
[v5] Import files routes into routes/index and create a global group

### DISCUSSION THREAD

**Discussion by jairochaves2** [adonisjs/core#3113] [Help] [answered]
I would like to create a group with a prefix that encompasses all routes/groups, but I'm putting each route into a different file, and importing it into a route/index.ts.

```ts
// project route file
const group = Route.group(() => {
  Route.resource('projects', 'Project/MainController').apiOnly()
  Route.get('project/medias', 'Project/Medias.index')
}).prefix('p')
```
```ts
//index route
Route.group(()=>{
    import('./project.ts')
}).prefix('api/v1')
```
I would like the generated route to be something like:

`/api/v1/p/projects`

**Comment by CodeRogers** [comment, marked as answer]
You can define your Route and export as function like these: 
```typescript
// Using resource:
import Route from '@ioc:Adonis/Core/Route'

export default function userRoutes() {
  Route.resource('users', 'UsersController').apiOnly()
}

// Using methods:
import Route from '@ioc:Adonis/Core/Route'

export default function authenticationRoutes() {
  Route.post('login', 'LoginController.store')
  Route.post('logout', 'LogoutController.store').middleware('auth')
}

// Using group: 
import Route from '@ioc:Adonis/Core/Route'

export default function companyRoutes() {
  Route.group(() => {
    Route.get('companies/user', 'Companies/SearchController.index')
    Route.resource('companies', 'Companies/CompaniesController').apiOnly().middleware({'*': 'bouncer: adm'})
  }).middleware(['auth'])
}
```
All the middlewares still working as they should  
  
Then you just need to import the functions in the /routes/index.ts file:
```typescript
import Route from '@ioc:Adonis/Core/Route'
import Env from '@ioc:Adonis/Core/Env'

import authenticationRoutes from './authentication'
import companyRoutes from './company'
import userRoutes from './user'

Route.group(() => {
  companyRoutes()
  userRoutes()
  authenticationRoutes()
}).prefix('your/prefix/here')
```

**Comment by rutonyru** [comment]
Hello,
Since I don't want to lengthen my router.ts with all the route definitions for all my entities, I'm using the concept described above to split them up into entity-specific route files.
I don't know if this is a good idea (I hope it is ;) ). My dev colleague (who uses express) strongly recommended it...
In my case, it's an adaptation that works in adonisjs 6...

**Comment by MrAhmedElkady** [comment]
Hello, I have an update for Adonis version 6.
You can add a file using the method mentioned above or by modifying the **adonisrc** file.

In the **preloads** section, you can see the router being called.
It can be modified as follows:

```
preloads: [
    () => import('#start/routes'),
    () => import('#start/api'),
    () => import('#start/kernel')

```

Now, we have a new file called api create api.ts in start:
```
router.group(() => {

    router.get('me', async ({ auth, response }) => {
        try {
            const user = auth.getUserOrFail()
            return response.ok(user)
        } catch (error) {
            return response.unauthorized({ error: 'User not found' })
        }
    })
    .use(middleware.auth({
        guards: ['api']
    }))
    
    router.post('register', [RegistersController, 'store']).as("register")
    router.post('login', [LoginController, 'store']).as("login")

}).prefix('/api')
```



Note
I am not sure if this will work with version 5, but you can try it by modifying the **.adonisrc.json** file in the same way mentioned above and let me know if it works for you! :)
And don't forget to use v5 codes in the **api** file.
